src/backend/joanie/core/models/products.py

```diff
@@ -1483,3 +1493,82 @@ class Meta:
 
     first_name = models.CharField(_("first name"), max_length=255)
     last_name = models.CharField(_("last name"), max_length=255)
+
+
+class Discount(BaseModel):
+    """
+    Discount model allows to define a discount that can be associated to a product.
+    """
+
+    class Meta:
+        db_table = "joanie_discount"
+        verbose_name = _("Discount")
+        verbose_name_plural = _("Discounts")
+        ordering = ["created_on"]
+        constraints = [
+            models.CheckConstraint(
+                check=models.Q(rate__isnull=False) | models.Q(amount__isnull=False),
+                name="discount_rate_or_amount_required",
+                violation_error_message="Discount rate or amount is required.",
+            ),
+            models.CheckConstraint(
+                check=models.Q(rate__isnull=True) | models.Q(amount__isnull=True),
+                name="discount_rate_and_amount_exclusive",
+                violation_error_message="Discount rate and amount are exclusive.",
+            ),
+        ]
+
+    amount = models.PositiveSmallIntegerField(
+        _("amount"),
+        help_text=_("Discount amount"),
+        null=True,
+        blank=True,
+    )
+    rate = models.PositiveSmallIntegerField(
+        _("rate"),
+        help_text=_("Discount rate"),
+        null=True,
+        blank=True,
+    )
+    product = models.OneToOneField(
+        to=Product,
+        verbose_name=_("product"),
+        related_name="discount",
+        on_delete=models.CASCADE,
+        null=True,
+    )
```

Comment:
I wondier if it should not rely on a `CourseProductRelation` instead of a `Product`. Like we did for `OrderGroup`
A `Product` can be think as a factory and `CourseProcutRelation` as an instance of this factory.

Reply:
I don't get the value here… I feel like it just adds complexity, but I may miss something.

Reply:
As discussed IRL, we remove the relation for now.